res/skins/QMLDemo/InfoBarButton.qml

```diff
@@ -0,0 +1,71 @@
+import "." as Skin
+import Mixxx 0.1 as Mixxx
+import QtGraphicalEffects 1.12
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+import "Theme"
+
+AbstractButton {
+    id: root
+
+    required property string group
+    required property string key
+    property alias foreground: foreground.data
+    property color normalColor: Theme.buttonNormalColor
+    required property color activeColor
+    property color pressedColor: activeColor
+
+    states: [
+        State {
+            name: "pressed"
+            when: root.pressed
+
+            PropertyChanges {
+                target: colorOverlay
+                color: root.pressedColor
+            }
+
+        },
+        State {
+            name: "active"
+            when: root.checked && !root.pressed
+
+            PropertyChanges {
+                target: colorOverlay
+                color: root.activeColor
+            }
+
+        },
+        State {
+            name: "inactive"
+            when: !root.checked && !root.pressed
+
+            PropertyChanges {
+                target: colorOverlay
+                color: root.normalColor
+            }
+
+        }
+    ]
+
+    Mixxx.ControlProxy {
+        group: root.group
+        key: root.key
+        value: root.checked || root.down
+    }
+
+    Item {
```

Comment:
```suggestion
    foreground: Item {
```
and then expose the foreground as a property? maybe it would even be possible to use an alias property?

Reply:
I don't this thats possible but I'll have a look. Generally, only the stuff in the `Mixxx` directory is meant to be reused by other skins, so I only added customization properties where actually needed.

Reply:
Thats understandable, I'm just suggesting it because you are actually exposing the foreground on L13 if I understand correctly.

Reply:
Now I get what you mean.

Your original suggestion won't work, this tries to assign the foreground property which doesn't exist (and If I declare it, QML doesn't know where to display it, because it's not part of the body).

And If I make foreground an alias property directly, the property is read-only.

Reply:
Mhmm right, how did I not notice that this would clash with the regular property assignment syntax...
SixtyFPS supports this using `id := Object {...}` syntax instead of having the id declared as a property on the object.
I wish that was possible because it would allow to shorten the creation of simple objects on the lines of `playCO := Mixxx.createCO(group, "play")` instead of requiring 3 times the amount of code.